src/utils/formValidationOptions.ts

```diff
@@ -11,12 +12,23 @@ type RegisterOptions = Partial<{
   validate: Validate | Record<string, Validate>
 }>
 
-export const textFieldValidation: (
-  name: string,
-  minLength: number,
-  maxLength: number,
+type TextValidationArgs = {
+  name: string
+  maxLength: number
+  minLength?: number
   required?: boolean
-) => RegisterOptions = (name, minLength, maxLength, required = false) => ({
+  pattern?: RegExp
+  message?: string
```

Comment:
Maybe `patternMessage` or something alike? This could hint that this is a generic message